### PR TITLE
Compile tracee submodule against local libbpfgo

### PR DESCRIPTION
**Pull request by grantseltzer**
This adds a make target to compile a tracee submodule against local libbpfgo. Since tracee is libbpfgo's most notable consumer, it's a good smoke test to make sure local changes don't break tracee. Once e2e tests are in a good spot we can run those from here as well.

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
I'm not sure I like this approach @grantseltzer. I don't think we should include tracee as a submodule of libbpfgo.

I mean, we can automate a test (shell ?) to clone the repo, test things out in an git-ignored directory... like a selftest... but having it as a submodule doesn't seem like a good idea to me.

Check out `3rdparty/btfhub.sh` file in tracee for an idea how to clone and keep an ignored directory updated (for btfhub in that case). Maybe we could do the same for this test. 

I just don't want to disturb other projects who have nothing to do with tracee. WDYT ?

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
I'm not sure I like this approach @grantseltzer. I don't think we should include tracee as a submodule of libbpfgo.

I mean, we can automate a test (shell ?) to clone the repo, test things out in an git-ignored directory... like a selftest... but having it as a submodule doesn't seem like a good idea to me.

Check out 3rdparty/btfhub.sh file in tracee for an idea how to clone and keep an ignored directory updated (for btfhub in that case). Maybe we could do the same for this test.

I just don't want to disturb other projects who have nothing to do with tracee. WDYT ?

**Comment by grantseltzer** [comment]
> I'm not sure I like this approach @grantseltzer. I don't think we should include tracee as a submodule of libbpfgo.
> 
> I mean, we can automate a test (shell ?) to clone the repo, test things out in an git-ignored directory... like a selftest... but having it as a submodule doesn't seem like a good idea to me.
> 
> Check out `3rdparty/btfhub.sh` file in tracee for an idea how to clone and keep an ignored directory updated (for btfhub in that case). Maybe we could do the same for this test.
> 
> I just don't want to disturb other projects who have nothing to do with tracee. WDYT ?

That makes sense, submodule is tying it closer together, i'll redo it as just cloning in a script

File: testing/compile-tracee.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+TRACEE_DIR="${BASEDIR}/testing/tracee"
+RETURN_CODE=0
+
+CMDS="rsync git cp rm mv"
+for cmd in ${CMDS}; do
+    command -v $cmd 2>&1 >/dev/null || die "cmd ${cmd} not found"
+done
+
+branch_clean() {
+    cd ${1} || die "could not change dirs"
```
Comment (rafaeldtinoco) [inline testing/compile-tracee.sh:23]:
```
++ basename /home/rafaeldtinoco/work/ebpf/libbpfgo/testing/tracee
+ die 'tracee not a repo dir'
+ echo tracee not a repo dir
tracee not a repo dir
```

This is not right, you are CWDing to ./testing/tracee and it does not contain Readme.md file.

File: testing/compile-tracee.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -e
+
+#
+# This script downloads and updates the aquasecurity/tracee repository
+#
```
Comment (rafaeldtinoco) [inline testing/compile-tracee.sh:5]: nit: there is an extra space after # here.

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]

File: testing/compile-tracee.sh (hunk 1 above)
Comment (grantseltzer) [reply testing/compile-tracee.sh:23]: I don't understand what you mean, doesn't it?

Comment (rafaeldtinoco) [reply testing/compile-tracee.sh:23]: Nm, my local tree had garbage, sorry.

**Review by rafaeldtinoco — APPROVED** [review]
LGTM

**Pull request merged**